tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1251,5 +1251,31 @@ def test_assertion_is_failure(dut):
     assert False
 
 
+@cocotb.test(expect_error=True)
+def test_timeout_func_fail(dut):
+    @cocotb.coroutine
+    def fail():
+        yield Timer(10, 'ns')
+    yield cocotb.triggers.with_timeout(fail(), timeout_time=1, timeout_unit='ns')
```

Comment:
I'd prefer to write this as:
```
try:
    yield cocotb.triggers.with_timeout(fail(), timeout_time=1, timeout_unit='ns')
except cocotb.triggers.SimTimeoutError:
    pass
else:
    assert False, "Expected a timeout"
```
Because that proves that execution continues after a non-test-level timeout